tests/api/entities/type_lock.test.ts

```diff
@@ -0,0 +1,46 @@
+import 'should'
+import { getSomeWdEditionUri } from '#fixtures/entities'
+import { addClaim, getByUri, removeClaim } from '#tests/api/utils/entities'
+import { getAdminUser } from '#tests/api/utils/utils'
+import { shouldNotBeCalled } from '#tests/unit/utils/utils'
+
+describe('entities type lock', () => {
+  it('should reject non-dataadmin edits', async () => {
+    const uri = await getSomeWdEditionUri()
+    await addClaim({ uri, property: 'invp:P3', value: 'work' })
+    .then(shouldNotBeCalled)
+    .catch(err => {
+      err.statusCode.should.equal(403)
+      err.body.status_verbose.should.equal("editing property requires admin's rights")
+    })
+  })
+
+  it('should reject invalid types', async () => {
```

Comment:
What about having a test to assert that an `invp:P3` cannot be added if it is the same as the current type:

```
  it('should reject when typed-locked entities type is the same as the current entity type', async () => {
```

Reply:
There might be cases where that would be annoying: imagine that an item identified as having an unstable type, because it's has two `wdt:P31` values